### ISSUE TITLE
Add support for bad char/bad bytes comparisons similar to what mona has

### ISSUE THREAD

**Issue by nicolaipre** [pwndbg/pwndbg#1028]
As discussed in the pwndbg Discord server with disconnect3d, I am writing this FR.

It would be nice if there was a bad char comparison function in pwndbg that is equivalent to monas `cmp` command:
```
!mona cmp -f C:\mona\vulnserver\bytearray.bin -a <badchars memory address>
```
In the example above, the bytearray.bin contain bytes from 0x00-0xff / the payload you send. The cmp command will then compare this byte array with a specific memory address (or better yet, search through the memory for a match)

I looked at the encoder https://docs.pwntools.com/en/stable/encoders.html where you can specify bytes you want to avoid, but it would be nice if there exists something to detect them as well. 

The purpose of this would allow you to compare a certain array of bytes with a specific section in memory at a given address. 

Imagine you send a payload of 256 bytes (0x00-0xff), then i.e 0x0b is the only bad byte, the command would compare your byte array/payload to what is found in memory and highlight the mismatch/bad bytes in a (red?) color to make it easy to spot. 

I believe this could be very helpful when debugging payload shellcode that is not working as intended in cases which bad bytes are the reason behind it.

**Comment by disconnect3d** [comment]
I guess this could be some feature for `hexdump` command like `hexdump --hightlight=32,31,30,4?` where `?` means any nibble?

**Comment by nicolaipre** [comment]
> I guess this could be some feature for `hexdump` command like `hexdump --hightlight=32,31,30,4?` where `?` means any nibble?

Yeah possibly, but it would also be nice to be able to read the bytes to be compared from a local file and a byte string so you have both/all options.

Perhaps something like this, where the bytes that differ gets colored?
```shell
# highlight certain opcodes at address
pwndbg> hexdump --highlight=32,31,30,4? [address] [count]

# highliht difference in hex string and memory area at address
pwndbg> hexdump --cmp --bytes="deadbeef" [address] [count]

# highlight difference in file (eg: payload.bin) and memory area at address
pwndbg> hexdump --cmp --file="./payload.bin" [address] [count]
```

Although slightly different where a Window opens instead of using coloring (which I thought would be more suitable for pwndbg), here is a GIF from a [blog post](https://fluidattacks.com/blog/vulnserver-lter-seh/) demonstrating how it is implemented in Mona, showing the "Corruption after X bytes" and bad bytes based on the comparison:

![image](https://res.cloudinary.com/fluid-attacks/image/upload/v1620331205/blog/vulnserver-lter-seh/mona-cmp1_jgfekz.gif)
